base/essentials.jl

```diff
@@ -166,13 +166,16 @@ true
 """
 function convert end
 
+convert(::Type{Union{}}, x::Union{}) = throw(MethodError(convert, (Union{}, x)))
```

Comment:
This is my favorite method.

Reply:
Yes, this PR has been an exercise in "how on earth can inference discover an ambiguity *here*?"